ckanext/example_idatastorebackend/example_sqlite.py

```diff
@@ -23,15 +28,14 @@ def _get_engine(self):
     def _insert_records(self, table, records):
         if len(records):
             for record in records:
-                self._get_engine().execute(
-                    u'INSERT INTO "{0}"({1}) VALUES({2})'.format(
+                self.execute(
+                    'INSERT INTO "{0}"({1}) VALUES({2})'.format(
                         table,
                         u', '.join(record.keys()),
-                        u', '.join(['?'] * len(record.keys()))
+                        u', '.join([f":{key}" for key in record])
```

Comment:
sql injection from user-provided column names?

Reply:
haven't paid much attention to this backend because it's just an example, but you are right. I'll update these two injection opportunities

Reply:
PS I just love SQL constructor from SQLALchemy:
```python
sql = sa.insert(
  sa.table(table, *map(sa.column, record))
).values(record)
self.execute(sql)
```